src/MusicStore/Areas/Admin/Controllers/StoreManagerController.cs

```diff
@@ -45,27 +48,43 @@ public StoreManagerController(MusicStoreContext dbContext)
             var cacheKey = GetCacheKey(id);
 
             Album album;
-            if (!cache.TryGetValue(cacheKey, out album))
+
+            if (AppSettings.Value.CacheTimeout > 0)
+            {
+                if (!cache.TryGetValue(cacheKey, out album))
+                {
+                    album = await DbContext.Albums
+                            .Where(a => a.AlbumId == id)
+                            .Include(a => a.Artist)
+                            .Include(a => a.Genre)
+                            .FirstOrDefaultAsync();
+
+                    if (album != null)
+                    {
+                        //Remove it from cache if not retrieved in last 10 minutes.
+                        cache.Set(
+                            cacheKey,
+                            album,
+                            new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromSeconds(AppSettings.Value.CacheTimeout)));
+                    }
+                }
+            }
+            else
             {
                 album = await DbContext.Albums
                         .Where(a => a.AlbumId == id)
                         .Include(a => a.Artist)
                         .Include(a => a.Genre)
                         .FirstOrDefaultAsync();
-
-                if (album != null)
-                {
-                    //Remove it from cache if not retrieved in last 10 minutes.
-                    cache.Set(
-                        cacheKey,
-                        album,
-                        new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(10)));
-                }
             }
 
+
             if (album == null)
             {
-                cache.Remove(cacheKey);
+                if (AppSettings.Value.CacheTimeout > 0)
```

Comment:
Indeed. It looks like the expression `AppSettings.Value.CacheTimeout > 0` actually means `should we use the cache?` but it's not explicitly mentioned, in comments, or in a utility method. This is nitpicking, but consider making this more clear.